detectron2/structures/masks.py

```diff
@@ -343,3 +349,12 @@ def crop_and_resize(self, boxes: torch.Tensor, mask_size: int) -> torch.Tensor:
         if len(results) == 0:
             return torch.empty(0, mask_size, mask_size, dtype=torch.bool, device=device)
         return torch.stack(results, dim=0).to(device=device)
+
+    def area(self):
```

Comment:
could you add some docs?
And since `Boxes.area()` (and many other methods) returns a tensor, it's good to be consistent and returns a tensor.

Reply:
> could you add some docs?

I assume pointing to the stackoverflow in the `polygon_area` function is not sufficient, is it?

> And since `Boxes.area()` (and many other methods) returns a tensor, it's good to be consistent and returns a tensor.

Valid point

Reply:
Hey @ppwwyyxx,

This is a bit odd, if you look at the `PolygonMask` definition, then you can see the signature of the function is: `_make_array(t: Union[torch.Tensor, np.ndarray]) -> torch.Tensor`

but it converts it into a numpy array, no matter what was the input:
```
if isinstance(t, torch.Tensor):
    t = t.cpu().numpy()
    return np.asarray(t).astype("float64")
```

Reply:
what's the point of switching to numpy array?

Reply:
documentation fixed as requested in [1ae26f1](https://github.com/facebookresearch/detectron2/pull/175/commits/1ae26f127c3d9a184b7881a8832273fc2d8862b1)

Reply:
> what's the point of switching to numpy array?

because they are never used on GPUs and never had gradients, neither is likely to happen in the future as well. And converting a list of arrays to torch Tensor is sometimes surprisingly slow (if not done carefully).

For this reason I think the TODOs below can be removed.